src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs

```diff
@@ -410,6 +410,27 @@ private static string GetParameterText(string parameterName)
                                                   commandRedirection, context);
                 }
 
+                var cmdletInfo = commandProcessor?.CommandInfo as CmdletInfo;
+                if (cmdletInfo?.ImplementingType == typeof(OutNullCommand))
+                {
+                    var commandsCount = pipelineProcessor.Commands.Count;
+                    if (commandsCount == 1)
+                    {
+                        // Out-Null is the only command, bail without running anything
+                        return;
+                    }
+
+                    // Out-Null is the last command, rewrite command before Out-Null to a null pipe, but
+                    // only if it didn't redirect anything, e.g. `Get-Stuff > o.txt | Out-Null`
```

Comment:
The only reason I check for redirection was to avoid knowingly breaking that useless scenario, I don't see a reason to add more code to optimize it when it should never really happen anyway.